autofit/optimize/non_linear/nested_sampling/dynesty.py

```diff
@@ -0,0 +1,527 @@
+
+        self.vol_dec = self.config("vol_dec", float)
+        self.vol_check = self.config("vol_check", float)
+        self.walks = self.config("walks", int)
+        self.facc = self.config("facc", float)
+        self.slices = self.config("slices", int)
+        self.fmove = self.config("fmove", float)
+        self.max_move = self.config("max_move", int)
+
+        logger.debug("Creating DynestyStatic NLO")
+
+    def copy_with_name_extension(self, extension, remove_phase_tag=False):
+        """Copy this instance of the dynesty non-linear search with all associated attributes.
+
+        This is used to set up the non-linear search on phase extensions."""
+        copy = super().copy_with_name_extension(
+            extension=extension, remove_phase_tag=remove_phase_tag
+        )
+        copy.iterations_per_update = self.iterations_per_update
+        copy.bound = self.bound
+        copy.sample = self.sample
+        copy.update_interval = self.update_interval
+        copy.bootstrap = self.bootstrap
+        copy.enlarge = self.enlarge
+        copy.vol_dec = self.vol_dec
+        copy.vol_check = self.vol_check
+        copy.walks = self.walks
+        copy.facc = self.facc
+        copy.slices = self.slices
+        copy.fmove = self.fmove
+        copy.max_move = self.max_move
+
+        return copy
+
+    def _simple_fit(self, model: AbstractPriorModel, fitness_function) -> Result:
+        """
+        Fit a model using Dynesty and a function that returns a likelihood from instances of that model.
+
+        Dynesty is not called once, but instead called multiple times every iterations_per_update, such that the
+        sampler instance can be pickled during the model-fit. This allows runs to be terminated and resumed.
+
+        Parameters
+        ----------
+        model
+            The model which generates instances for different points in parameter space. This maps the points from unit
+            cube values to physical values via the priors.
+        fitness_function
+            A function that fits this model to the data, returning the likelihood of the fit.
+
+        Returns
+        -------
+        A result object comprising the best-fit model instance, likelihood and an *Output* class that enables analysis
+        of the full chains used by the fit.
+        """
+        dynesty_output = DynestyOutput(model=model, paths=self.paths)
+
+        if os.path.exists("{}/{}.pickle".format(self.paths.chains_path, "dynesty")):
+
+            with open(
+                "{}/{}.pickle".format(self.paths.chains_path, "dynesty"), "rb"
+            ) as f:
+                dynesty_sampler = pickle.load(f)
+
+        else:
+
+            dynesty_sampler = self.sampler_fom_model_and_fitness(
+                model=model, fitness_function=fitness_function
+            )
+
+        # These hacks are necessary to be able to pickle the sampler.
+
+        dynesty_sampler.rstate = np.random
+        pool = Pool(processes=1)
```

Comment:
A single process pool?

Reply:
For now, parallel NLS's is something we should discuss next week.